packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -208,17 +244,13 @@ export abstract class FluidDataStoreContext
 	public readonly gcTombstoneEnforcementAllowed: boolean;
 
 	/** If true, this means that this data store context and its children have been removed from the runtime */
-	private deleted: boolean = false;
+	protected deleted: boolean = false;
 
 	public get attachState(): AttachState {
 		return this._attachState;
 	}
 
 	public get IFluidDataStoreRegistry(): IFluidDataStoreRegistry | undefined {
-		assert(
-			this.channel !== undefined,
-			0x8f3 /* This should be called after the channel is created, when the registry is populated */,
-		);
 		return this.registry;
```

Comment:
why is the assert removed?

Reply:
This is an interesting case. I've added assert in my previous PR as a way to simplify debugging (as when the flow was wrong, we were getting here before registry was initialized).
With nested channels, if L1 creates L2 from L1's initialization, then L1 is not yet fully setup (we sit in attachRuntime / bindRuntime). But in order for L2 to be created, it will need to ask its parent (L1) for a factory, and thus we get here before bindRuntime() finished initialization for L1 (by doing this.channel = channel)..

There are probably ways to add some assert and allow this scenario (like this.registry !== undefined || this.channel !== undefined), but not sure if it's more helpful, or more misleading.

Let me poke at it, if I can find some useful way to add it without breaking scenarios, then I'll add it in next PR.